### PR TITLE
Create ExtendedJsonRpcApi and add `gettxhistory` and `getnodestate`

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
- Creates ExtendedJsonRpcApi including a switch to turn it on and off from `np-api-server` as discussed here: https://github.com/CityOfZion/neo-python/pull/614#issuecomment-423501873
- Adds `gettxhistory` (passing build from this PR: https://github.com/CityOfZion/neo-python/pull/614)
- Adds `getnodestate` (passing build from this PR: https://github.com/CityOfZion/neo-python/pull/598)

I am delaying adding the methods from this PR because I had yet to achieve a passing build: https://github.com/CityOfZion/neo-python/pull/573

**How did you solve this problem?**
Trial and Error

**How did you make sure your solution works?**
Unittest and personal testing with my local RPC node

**Are there any special changes in the code that we should be aware of?**
No

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: neo/bin/api_server.py [hunk 1]
```diff
@@ -245,9 +248,17 @@ def loopingCallErrorHandler(error):
...
+        api_server_rpc = JsonRpcApi(args.port_rpc, wallet=wallet, extended=True)
+        endpoint_rpc = "tcp:port={0}:interface={1}".format(args.port_rpc, args.host)
+        endpoints.serverFromString(reactor, endpoint_rpc).listen(Site(api_server_rpc.app.resource()))
+#        reactor.listenTCP(int(args.port_rpc), server.Site(api_server_rpc))
+#        api_server_rpc.app.run(args.host, args.port_rpc)
+
+    elif args.port_rpc:
         logger.info("Starting json-rpc api server on http://%s:%s" % (args.host, args.port_rpc))
-        api_server_rpc = JsonRpcApi(args.port_rpc, wallet=wallet)
+        api_server_rpc = JsonRpcApi(args.port_rpc, wallet=wallet, extended=False)
```
[the remark this answers is not in the document]
Comment (jseagrave21) [reply neo/bin/api_server.py]: How to we override the `json_rpc_method_handler` while still retaining all the original functionality of `JsonRpcApi` and allowing for extended methods `getnodestate` that depend on `_init_`?

**Comment by jseagrave21** [comment]
@localhuman @ixje Could you provide further clarification concerning the desired end state and implementation of the `ExtendedJsonRpc`? My proposed method provides all original JSONRPC or original + extended JSONRPC functionality depending on user input when starting the server. Although there is some additional logic in `JsonRpcApi`, these changes seem necessary to me, especially for specific methods like `getnodestate`.

**Comment by localhuman** [comment]
I would extend `JsonRpcApi` like this:
```
class ExtendedJsonRpcApi(JsonRPCApi):

  def __init__(self, port, wallet=None):
     // do extended things
     super(self, ExtendedJsonRpcApi).__init__(port,wallet)

  def json_rpc_method_handler(self, method, params):
     // do extended things

    // then call super method to handle things implemented in base class
     super(self,ExtendedJsonRpcApi).json_rpc_method_handler(method,params)

```

**Comment by jseagrave21** [comment]
@localhuman Okay, this PR is ready for another review

**Comment by localhuman** [comment]
Much better, thanks for sticking with it! 🥇

**Pull request merged**